pyk/src/pyk/cterm/symbolic.py

```diff
@@ -255,7 +255,7 @@ def implies(
         if result.predicate is None:
             raise ValueError('Received empty predicate for valid implication.')
         ml_subst = self.kore_to_kast(result.substitution)
-        ml_pred = self.kore_to_kast(result.predicate) if result.predicate is not None else mlTop()
```

Comment:
Line 255 above guarantees that `result.predicate` is not `None` and the type checker agrees.